File: purchasely/example/ios/PurchaselyViewManager.m
//
//  NativeViewManager.m
//  example
//
//  Created by Chouaib Mounaime on 05/03/2024.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(PurchaselyViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(placementId, NSString)
RCT_EXPORT_VIEW_PROPERTY(completionCallback, RCTBubblingEventBlock)

@end

File: purchasely/example/ios/PurchaselyViewManager.swift
//
//  NativeViewManager.swift
//  example
//
//  Created by Chouaib Mounaime on 05/03/2024.
//

import Foundation

@objc (PurchaselyViewManager)
class PurchaselyViewManager: RCTViewManager {
 
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
 
  override func view() -> UIView! {
    return PurchaselyView()
  }
}
